ci/scripts/quick-bazel.sh

```diff
@@ -8,7 +8,82 @@
 # This doesn't install dependencies, but should otherwise behave the
 # same as what CI would do on a pull request
 
-bazel query "rdeps(//..., //hw:verilator)" | \
-            sed -e 's/^/-/' | \
-            xargs bazel test --nobuild_tests_only --test_tag_filters=-cw310,-verilator -- //...
+set -e -x
+
+DEFAULT="master"
+
+git config advice.detachedHead false
+
+ORIGINAL_BRANCH_NAME=$(git rev-parse --abbrev-ref HEAD 2>/dev/null || echo "")
+ORIGINAL_SHA=$(git rev-parse HEAD)
+
+mkdir -p bazel-results
+
+# If a test is moved or intentionally broken, tagging it lets CI know to skip
+# it when assessing the pr and the merge base.
+bazel query 'attr("tags", "broken", tests(//...))' > bazel-results/ci_run_ignore.txt
+
+summarize_tests () {
+    bazel query 'rdeps(//..., //hw:verilator)' > bazel-results/invokation_ignore.txt
+    cat bazel-results/ci_run_ignore.txt bazel-results/invokation_ignore.txt | \
+        sed -e 's/^/-/' | \
+        xargs bazel test --nobuild_tests_only --test_tag_filters=-broken,-cw310,-verilator -- //... | \
+        grep '//' | \
+        sed 's/ (cached) / / ; s/ in / / ; s/ [0-9]\+.[0-9]\+s// ; s/ \+/ /g' | \
+        sort > "$1"
+    return "${PIPESTATUS[0]}"
+}
+
+
+if summarize_tests bazel-results/newresult.txt
+then
+  exit 0
+fi
+
+git remote add upstream https://github.com/lowrisc/opentitan.git || true
+git fetch upstream $DEFAULT
+
+if [ "$(git merge-base HEAD upstream/$DEFAULT)" != "$(git rev-parse HEAD)" ]
+then
+  # If on a branch we want to compare to the merge-base with master
+  git checkout "$(git merge-base HEAD upstream/$DEFAULT)"
+else
+  # If on master we want to compare to the last tested ancestor, but that's
+  # fairly hard to review so let's leave that alone for now. We wouldn't act on
+  # it anyway.
+  echo "Test is failing but I can't compare it to a merge base because we're on upstream/$DEFAULT"
+  cat bazel-results/newresult.txt
+  exit 0
+fi
```

Comment:
This all looks much too magical for me. Is there a fundamental technical reason that we feel we must run failing tests in CI? If not, I suggest turning the broken tests off until they are fixed, fixing them, and then turning them on.

Look, ma! No scripting required! :-)